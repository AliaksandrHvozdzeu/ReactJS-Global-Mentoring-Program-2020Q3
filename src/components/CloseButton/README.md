#Close modal window button component
