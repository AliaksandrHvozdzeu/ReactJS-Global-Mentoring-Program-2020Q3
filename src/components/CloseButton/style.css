.close-button {
  position: absolute;
  right: 20px;
  top: 20px;
  background-color: transparent;
  color: transparent;
  border: transparent;
  outline: none;
  width: 2rem;
  height: 2rem;
}

.close-button::before {
  content: '';
  position: absolute;
  border-top: 2px solid var(--white);
  transform: rotate(45deg);
  left: 0;
  top: 50%;
  width: 2rem;
}

.close-button::after {
  content: '';
  position: absolute;
  border-top: 2px solid var(--white);
  transform: rotate(-45deg);
  left: 0;
  top: 50%;
  width: 2rem;
}
