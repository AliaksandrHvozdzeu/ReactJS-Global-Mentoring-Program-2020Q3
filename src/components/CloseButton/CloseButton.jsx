import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

export default function CloseButton({ closeAction }) {

  return (
    <>
      <button type="button" aria-label="close" className="close-button" onClick={closeAction}/>
    </>
  );

}
CloseButton.propTypes = {
  closeAction: PropTypes.func.isRequired,
};
