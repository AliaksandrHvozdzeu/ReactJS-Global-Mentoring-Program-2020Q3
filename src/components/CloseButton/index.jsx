import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

const CloseButton = ({ closeAction }) => (
  <button type="button" className="close-button" onClick={closeAction}/>
);

CloseButton.propTypes = {
  closeAction: PropTypes.func.isRequired,
};

export default CloseButton;
