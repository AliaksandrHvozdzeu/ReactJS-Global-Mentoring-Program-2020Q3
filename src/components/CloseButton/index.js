import CloseButton from './js/CloseButton';

export default CloseButton;
