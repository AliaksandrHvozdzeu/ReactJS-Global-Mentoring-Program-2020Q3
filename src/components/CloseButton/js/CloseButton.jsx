import React from 'react';
import PropTypes from 'prop-types';
import '../css/CloseButton.css';

export default function CloseButton({ closeAction }) {

  return (
    <button type="button"
            aria-label="close"
            className="close-button"
            onClick={closeAction}/>
  );

}
CloseButton.propTypes = {
  closeAction: PropTypes.func.isRequired,
};
