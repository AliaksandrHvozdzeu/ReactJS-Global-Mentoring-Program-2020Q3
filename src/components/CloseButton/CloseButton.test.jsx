import React from 'react';
import rerender from 'react-test-renderer';
import CloseButton from './js/CloseButton';

describe('when CloseButton', () => {
  test('then snapshot created', () => {
    const closeAction = () => {};
    const component = rerender.create(<CloseButton closeAction={closeAction}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
