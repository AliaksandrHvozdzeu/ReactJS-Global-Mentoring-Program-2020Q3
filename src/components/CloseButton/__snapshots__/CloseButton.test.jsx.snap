// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when CloseButton then snapshot created 1`] = `
<button
  aria-label="close"
  className="close-button"
  onClick={[Function]}
  type="button"
/>
`;
